src/PrestaShopBundle/Resources/config/services/core/addon.yml

```diff
@@ -29,3 +29,19 @@ services:
           - "@filesystem"
           - "@prestashop.core.admin.lang.repository"
           - "@prestashop.translation.theme.exporter"
+
+    prestashop.core.addon.theme.theme_manager_builder:
+        class: PrestaShop\PrestaShop\Core\Addon\Theme\ThemeManagerBuilder
+        arguments:
+            - '@=service("prestashop.adapter.legacy.context").getContext()'
+            - '@prestashop.adapter.legacy_db'
+
+    prestashop.core.addon.theme.theme_manager:
+        class: PrestaShop\PrestaShop\Core\Addon\Theme\ThemeManager
+        factory: 'prestashop.core.addon.theme.theme_manager_builder:build'
+
+    prestashop.core.addon.theme.theme.page_layouts_customizer:
+        class: PrestaShop\PrestaShop\Core\Addon\Theme\ThemePageLayoutsCustomizer
+        arguments:
+           - '@=service("prestashop.adapter.legacy.context").getContext().shop.theme'
```

Comment:
What if the shop theme gets changed during the http request handling by php ?
Is it gonna do the following scenario:
1) SF container loads, gets the shop theme A from context
2) controller gets called, handles http requests and replaces shop theme A, now it's shop theme B
3) another processing is performed which requires to use se Customizer but customization is performed using theme A and not the latest, theme B

Reply:
hmm, i dont really know if thats an issue or maybe dont understand whole process.

if you were to inject `Context` you would do the same `$context->shop->theme` call and get the same result? But doing call directly in service declaration helps isolate `ThemePageLayoutsCustomizer` from legacy, am i missing something? :thinking:

Reply:
Let's go for this and change it later if required 👍